packages/iris-grid/src/IrisGridTableModel.js

```diff
@@ -460,20 +460,19 @@ class IrisGridTableModel extends IrisGridModel {
 
   getColumnsWithLayoutHints = memoize((columns, hints) => {
     if (hints) {
-      const frontColumns = new Array(hints.frontColumns.length);
-      const backColumns = new Array(hints.backColumns.length);
-      const middleColumns = [];
-
-      columns.forEach(col => {
-        if (hints.frontColumns.includes(col.name)) {
-          frontColumns[hints.frontColumns.indexOf(col.name)] = col;
-        } else if (hints.backColumns.includes(col.name)) {
-          backColumns[hints.backColumns.indexOf(col.name)] = col;
-        } else {
-          middleColumns.push(col);
-        }
-      });
-
+      const columnMap = new Map();
```

Comment:
As we briefly discussed on the phone, we only want the column to appear once in the final array, so if it's added to the front AND the back, we should handle that (or throw an error). Also, this doesn't warn if there's columns that don't even exist, it's just filtering them out (I think?). To make it easy... see if we can just throw an error and if that's handled correctly? Eg.
```
if (hints) {
  if (// Check if the same name exists in front and back columns ||  // Check a name is in front/back cols that isn't a valid column) {
    throw new Error('Layout hints are invalid');
  }
  
}
```

Reply:
Added a check for invalid column names. Duplicates seem handled fine (silently) at the API.